### PR TITLE
Fix bug with bad credentials errors message when connecting to Postgres

### PR DESCRIPTION
**Pull request by kenxu95**
## Describe your changes and why you are making these changes
Task: https://linear.app/aqueducthq/issue/ENG-1480/fix-bug-where-we-spuriously-call-an-error-unexpected

Fixed:
image

Authenticate is a little bit weird since for some integrations, it conflates with connector initialization, so I pulled it to the top and redirected all initialization and authentication errors to the same helpful error message.

## Related issue number (if any)
ENG-1480

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

**Review by saurav-c — APPROVED** [review]
LGTM

File: src/python/aqueduct_executor/operators/connectors/data/execute.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import sys
-from typing import Any
+from typing import Any, cast
```
Comment (likawind) [inline src/python/aqueduct_executor/operators/connectors/data/execute.py:2]: where is `cast` being used?

Comment (kenxu95) [reply src/python/aqueduct_executor/operators/connectors/data/execute.py:2]: oh that's a leftover from when I was debugging mypy, can remove

**Review by likawind — APPROVED** [review]
Just to confirm, error thrown after this change will also have the correct error type to be `user_fatal`?

**Comment by kenxu95** [comment]
Yep, and it'll be consistent across all data connectors now.

**Comment by kenxu95** [comment]
Skipping integration tests cus I just need to make an unused import removal.

**Pull request merged**